internal/services/pam/pam.go

```diff
@@ -269,6 +269,10 @@ func (s Service) IsAuthenticated(ctx context.Context, req *authd.IARequest) (res
 	}
 
 	access, data, err := broker.IsAuthenticated(ctx, sessionID, string(authenticationDataJSON))
+	if errors.Is(err, context.Canceled) {
+		log.Debugf(ctx, "Authentication for session %s was canceled", sessionID)
+		return &authd.IAResponse{Access: auth.Cancelled}, nil
+	}
```

Comment:
I think this fit better at lower level (`(b Broker) IsAuthenticated`), since it's also where we handle the cancellation currently, and that happens even when we receive the cancellation from PAM client side, no?

So that all the logic is in the same place, while this method looks more a simpler proxy to me.

Reply:
But @denisonbarbosa or @didrocks have better understanding of this part.

Reply:
I'm fine with that. I pushed a fixup commit. While trying it out, I noticed another case which could result in an error message when the authentication is canceled. Please take a look at that commit too.